src/subcommand/wallet/buy.rs

```diff
@@ -0,0 +1,278 @@
+use {
+  super::*,
+  crate::wallet::Wallet,
+  bitcoin::{
+    blockdata::{locktime::absolute::LockTime, witness::Witness},
+    psbt::Psbt,
+  },
+};
+
+#[derive(Serialize, Deserialize)]
+pub struct Output {
+  pub setup_txid: Option<Txid>,
+  pub purchase_txid: Txid,
+  pub total_fees: u64,
+  pub total_cost: u64,
+}
+
+const BUMP_SATS: u64 = 600;
+
+#[derive(Debug, Parser, Clone)]
+pub(crate) struct Buy {
+  pub psbt: String,
+  #[arg(long, help = "Use fee rate of <FEE_RATE> sats/vB")]
+  fee_rate: FeeRate,
+  #[arg(long, help = "Don't broadcast transactions.")]
+  pub(crate) dry_run: bool,
+}
+
+impl Buy {
+  pub(crate) fn run(self, options: Options) -> SubcommandResult {
+    let index = Index::open(&options)?;
+    index.update()?;
+
+    let client = options.bitcoin_rpc_client_for_wallet_command(false)?;
+    let unspent_outputs = index.get_unspent_outputs(Wallet::load(&options)?)?;
+    let inscriptions = index.get_inscriptions(&unspent_outputs)?;
+    let chain = options.chain();
+
+    let all_inscription_outputs = inscriptions
+      .keys()
+      .map(|satpoint| satpoint.outpoint)
+      .collect::<HashSet<OutPoint>>();
+
+    let spendable_utxos = unspent_outputs
+      .iter()
+      .filter_map(|e| match all_inscription_outputs.get(e.0).is_some() {
+        true => None,
+        false => Some(e.clone()),
+      })
+      .collect::<Vec<_>>();
+
+    let mut fund_utxos = spendable_utxos
+      .iter()
+      .filter_map(|e| match e.1.to_sat() == BUMP_SATS {
+        true => None,
+        false => Some((e.0.clone(), e.1.clone())),
+      })
+      .collect::<Vec<_>>();
+
+    let mut bump_utxos = spendable_utxos
+      .iter()
+      .filter_map(|e| match e.1.to_sat() == BUMP_SATS {
+        true => Some((e.0.clone(), e.1.clone())),
+        false => None,
+      })
+      .collect::<Vec<_>>();
+
+    let mut signed_setup_tx = None;
+
+    let mut total_fees = 0;
+
+    if bump_utxos.len() < 2 {
```

Comment:
Instead of going with the bump pattern that market places use, it would be great if there was an option for the user to use any 2 random UTXOs for the first 2. That would allow UTXO consolidation as a nice byproduct of a purchase.